### PR TITLE
Added Bluecoat ProxySG Support

### PR DESCRIPTION
**Pull request by NerdBlender**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

**Comment by laf** [comment]
You've not signed the contributors agreement. http://docs.librenms.org/General/Contributing/

**Comment by NerdBlender** [comment]
Please see #5169

File: includes/definitions.inc.php [hunk 1]
```diff
@@ -3364,6 +3379,11 @@
 $config['device_types'][$i]['type'] = 'appliance';
 $config['device_types'][$i]['icon'] = 'appliance.png';
 
+$i++;
+$config['device_types'][$i]['text'] = 'Proxy';
```
Comment (laf) [inline includes/definitions.inc.php]: @librenms/reviewers Are we happy to add this additional device_type in?

File: includes/definitions.inc.php [hunk 2]
```diff
@@ -2758,6 +2758,19 @@
 $config['os'][$os]['over'][2]['graph'] = 'device_mempool';
 $config['os'][$os]['over'][2]['text']  = 'Memory Usage';
 
+// Bluecoat SGOS
+$os = 'sgos';
+$config['os'][$os]['text']             = 'BlueCoat ProxySG';
+$config['os'][$os]['type']             = 'network';
+$config['os'][$os]['ifname']           = 1;
+$config['os'][$os]['icon']             = 'sgos';
```
Comment (laf) [inline includes/definitions.inc.php]: Icon should be called bluecoat.

Comment (laf) [inline includes/definitions.inc.php]: And rename the icon itself to bluecoat.png

File: includes/discovery/os/sgos.inc.php [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+if (!$os) {
```
Comment (laf) [inline includes/discovery/os/sgos.inc.php]: This isn't needed, you can remove the `if (!$os) {` check.

File: includes/discovery/sensors/fanspeeds/sgos.inc.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
+// BlueCoat ProxySG Fanspeeds
+if ($device['os'] == 'sgos') {
+        echo 'ProxySG ';
```
Comment (laf) [inline includes/discovery/sensors/fanspeeds/sgos.inc.php]: Too much white space, 4 spaces should be used for indentation

File: includes/discovery/sensors/fanspeeds/sgos.inc.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        $fan_index = 0;
+    for ($index = 21; $index < 39; $index++) { //Proxy SG Fan OID end in 21-38
+          $fanstatus_oid   = ".1.3.6.1.4.1.3417.2.1.1.1.1.1.6.$index";
```
Comment (laf) [inline includes/discovery/sensors/fanspeeds/sgos.inc.php]: Too much white space, 4 spaces should be used for indentation

File: includes/definitions.inc.php (hunk 1 above)
Comment (murrant) [inline includes/definitions.inc.php]: I agree that we don't want the list to get too long.  Why would we use "Proxy" over "Network" here?

**Pull request merged**